DSC Resource/JustEnoughAdministration/JustEnoughAdministration.psm1

```diff
@@ -216,13 +216,17 @@ class JeaEndpoint
                 Write-Verbose "UserDriveMaximumSize not equal: $($currentInstance.UserDriveMaximumSize)"
                 return $false
             }
-
-            $requiredGroupsHash = $this.ConvertStringToHashtable($this.RequiredGroups)
-            if(-not $this.ComplexObjectsEqual($this.ConvertStringToHashtable($currentInstance.RequiredGroups), $requiredGroupsHash))
-            {
-                Write-Verbose "RequiredGroups not equal: $(ConvertTo-Json $currentInstance.RequiredGroups -Depth 100)"
-                return $false
+            # Check for null required groups
+            if($currentInstance.RequiredGroups -ne $null)
```

Comment:
This condition is a bit too broad -- it ignores the case where the DSC config does not have any required groups, but the JEA endpoint registered on the system _does_ have required groups. That should fail the test since the endpoint does not match the desired state.

You could add an elsif to handle the case where the `$currentInstance.RequiredGroups -eq $null` but `$this.RequiredGroups -ne $null`, but I think it may be cleaner to update `ConvertStringToHashtable` and `ComplexObjectsEqual` to better handle null inputs.